lib/rust/ensogl/core/src/display/shape/primitive/shader/builder.rs

```diff
@@ -40,16 +40,17 @@ impl Builder {
     /// Returns the final GLSL code. If `pointer_events_enabled` is set to false, the generated
     /// shape will be transparent for pointer events and will pass them trough.
     #[profile(Detail)]
-    pub fn run<S: canvas::Draw>(shape: &S, pointer_events_enabled: bool) -> CodeTemplate {
+    pub fn run<S: canvas::Draw>(shape: &S, disable_pointer_events: &str) -> CodeTemplate {
         let mut canvas = Canvas::default();
         let shape_ref = shape.draw(&mut canvas);
         let shape_header = header("Shape Definition");
         canvas.add_current_function_code_line(format!("return {};", shape_ref.getter()));
         canvas.submit_shape_constructor("run");
         let shape_def = overload::allow_overloading(&canvas.to_glsl());
         let code = [GLSL_BOILERPLATE.as_str(), "", &shape_header, &shape_def].join("\n\n");
-        let main =
-            format!("bool pointer_events_enabled = {pointer_events_enabled};\n{FRAGMENT_RUNNER}");
+        let main = format!(
+            "bool pointer_events_enabled = ({disable_pointer_events}) == 0.0;\n{FRAGMENT_RUNNER}"
+        );
```

Comment:
Why enable/disable pointer events is resolved to float instead to `bool`? I've seen no explanation.

Reply:
Attribs can't be bool in GLSL

Reply:
> Attribs can't be bool in GLSL

So let's put a short comment somewhere.